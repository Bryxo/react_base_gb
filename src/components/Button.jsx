import React from "react";

export const Button = (props) => {
  return <button className="btn_published" onClick={props.click}>{props.name}</button>;
}