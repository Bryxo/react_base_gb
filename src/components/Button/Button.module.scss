.btn_published {
   margin: 20px 0 20px 0;
   border: none;
   cursor: pointer;
   background-color: rgb(232, 153, 56);
   border-radius: 5px;
}