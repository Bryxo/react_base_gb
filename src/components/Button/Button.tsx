import React, { FC } from 'react';
//import Button from 'Button';
import style from './Button.module.scss';

interface ButtonProps {
  name:string;
  // disabled: boolean;
  // onButtonClick?: () => void;
}

export const Button: FC<ButtonProps> = (props) => {
  return (
    //кнопка без библиотеки
    // <button className="btn_published" type="submit">
    //   {props.name}
    // </button>

    <button
      //variant="contained"
      type="submit"
      className={style.button}
    >
      {props.name}
    </button>

  );
};
