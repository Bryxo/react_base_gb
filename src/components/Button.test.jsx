import React from 'react';
import { Button } from './Button';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Button', () => {
  it('render component', () => {
    render(<Button />);
  });
  it('render component with text', () => {
    render(<Button />);
    expect(screen.getByText('Опубликовать')).toBeInTheDocument();
  });
  it('render with snapshot', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
