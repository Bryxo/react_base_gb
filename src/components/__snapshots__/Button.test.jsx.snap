// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn_published"
    type="submit"
  >
    Опубликовать
  </button>
</DocumentFragment>
`;
